subjects/zappy/README.md

```diff
@@ -0,0 +1,351 @@
+
+### Broadcast
+
+To send a message, the following command must be sent to the server by the client:
+
+```
+broadcast <text>
+```
+
+The server will send this line to all its clients:
+
+```
+message <K>,<text>
+```
+
+With K indicating the direction from where the sound comes.
+
+### Sound Transmission
+
+All the players listen to the broadcasts without knowing who broadcasts them, what they understand from it, is only the direction from which the sound comes and the message.
+
+The number of squares traversed by the sound before reaching the player indicates its direction.
+This numbering is done by assigning '1' to the square in front of the player, then counting down the squares around the player in a trigonometric (counterclockwise) direction.
+Remember, sound is a wave that moves in a linear manner and the world is round, so we'll choose the shortest path for sound between the transmitter and the player we calculated for.
+
+The following example indicates the sound trajectory that we must choose, as well as the numbers of the squares around the player. The player "A" is the source that sends te broadcast through the square 4 and "B" is the receptor witch receives the broadcast through square 2.
+
+![Example](./sound.png)
+
+If the broadcast is sent from the same box as the receiving player, he will receive the message from square 0.
+
+### The Server
+
+The server is the main engine that will interfere between the AI and the GUI. It will exchange all data with the AI and execute these instructions. At the same time, it send the information to the GUI. It can be written in one of these languages [C, C++, Rust].
+
+For this part, you must create a TCP server to make a connection between clients and manage the entire world.
+
+#### Server Instructions
+
+- Your server must generate all the resources in a random way following the rules mentioned above in the resources section.
+- Any request to your server must never hang forever. It must never block, and the `client` can be appropriately bounced if necessary.
+- The server executes the `client` requests in the order they are received.
+- The requests are buffered and the execution time of the command will only block the player concerned.
+- It's forbidden to use any `exec` functions to run another server.
+- It must stay available at all costs.
+
+Although you can use threads, we don't advise it. The use of multiplexing connections on your server is a better option.
+
+```console
+$ ./server
+	Usage: ./server -p <port> -x <width> -y <height> n <team> [<team>] [<team>] ... -c <nb> [-t <t>]
+    -p  is the port number
+    -x  is the width of the world
+    -y  is the height of the world
+    -n  is the name of the teams
+    -c  is the number of authorized clients per team
+    -t  is the time unit divider (the greater t is, the faster the game will go)
+```
+
+### The Client
+
+The Client can be written in one of these languages [C, C++, Rust, Js, Python]
```

Comment:
I see Go is not in the list anymore and C and C++ has been added, is there any reason for those changes or it was just a copy/paste that went wrong?

Reply:
according to Dav, zappy is C, C++ or Rust. 
Maybe we should remove Js and Python, what do you think?

Reply:
I like the idea having different language families for client and server. Since it's a group effort this can improve the possibility to share new knowledge in between the students.
Also students can decide better how to divide responsibilities and usually clients and servers are indeed in different languages.
A client in C/C++ (or even Rust) maybe won't make much sense since the focus will be on memory safety, fast execution and reliability... While for the client I would like to see students able to focus on nice algorithms to grow fast.

Still the decision is yours, everything works, I was just surprised to see that line changed

Reply:
I spoke to Dav again and he said he agrees with it being open language